### PR TITLE
Add gpu and kunlun chip support brief design doc.

### PR DESCRIPTION
**Pull request by tizhou86**
Hi all: 
I've just added gpu and kunlun chip support brief design doc, please take a look, thanks.#744

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
It is not clear what do you want to do in volcano. @k82cn Haven't we already supported ScalarResource scheduling?

**Comment by tizhou86** [comment]
> It is not clear what do you want to do in volcano. @k82cn Haven't we already supported ScalarResource scheduling?

@hzxuzhonghu ScalarResource cannot support fine grained gpu memory sharing, since there might be mutliple gpu cards on single machine, for example two card with 2g doesn't mean that there are 4g memory to be allocated to single pod, we need additional mechanism to handle that.

File: docs/design/gpu-and-kunlun-support.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        nvidia.com/gpu: 3
+        cpu: xxx
+        memory: xxx
+
+```
+
+
+#### Notes:
+
+Volcano adds startup parameters—gpu-type = ”volcano/gpu_k40_4, volcano/gpu_k40_16, volcano/gpu_p40_8, volcano/gpu_v100_8”
```
[the remark this answers is not in the document]
Comment (tizhou86) [reply docs/design/gpu-and-kunlun-support.md]: Sure, i'll update the doc for more implementation details later.

**Comment by hzxuzhonghu** [comment]
@tizhou86 Any update

**Comment by tizhou86** [comment]
> @tizhou86 Any update

I'm working on it now, will make a PR soon.

**Comment by hzxuzhonghu** [comment]
@tizhou86 Let's make it in this release. Any update?

**Comment by tizhou86** [comment]
> @tizhou86 Let's make it in this release. Any update?

Still working on the implementation and testing, when will this release end?

**Comment by hzxuzhonghu** [comment]
It is planned at the end of June 30.

**Comment by tizhou86** [comment]
> It is planned at the end of June 30.

It's ok to finish it in this release.

**Comment by hzxuzhonghu** [comment]
cool

**Comment by k82cn** [comment]
how about the design doc?

**Comment by tizhou86** [comment]
> how about the design doc?

Working on updating it.

File: docs/design/gpu-and-kunlun-support.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Specify the gpu request and limit of the above model in the submission job. At the same time, user need to specify the same number of nvidia.com/gpu resources (corresponding to the actual gpu setting)
+
+
+### Device Plugin changes
+
+Nvidia already support Device Plugin without GPU sharing mechanism. By providing a universal device plugin mechanism and a standard device API interface, it is opened through the feature gate, that is, configured --feature-gates = DevicePlugins = true. So we need to support KunLun Device Plugin and GPU sharing Device Plugin to meet our needs.
```
[the remark this answers is not in the document]
Comment (g-angstrom) [reply docs/design/gpu-and-kunlun-support.md:70]: Did you look at the node labeler work AMD did for it k8 device plug-in https://github.com/RadeonOpenCompute/k8s-device-plugin/blob/master/cmd/k8s-node-labeller/README.md 

The tool automatically labels nodes with GPU properties if a node has one or more AMD GPU installed.  Aka so you can support different GPU/Devices in the cluster ( but you need homogenous nodes)  and make it easier to select the appropriate GPU in a node  in the cluster 

Deployment

The Labeller needs to be run on all the nodes that are equipped with AMD GPU. It creates a Kubernetes DaemonSet, which runs a copy of a pod on all (or some) Nodes in the cluster. An example configuration is available here.

The Labeller currently creates node label for the following AMD GPU properties:

Device ID (-device-id)
VRAM Size (-vram)
Number of SIMD (-simd-count)
Number of Compute Unit (-cu-count)
Firmware and Feature Versions (-firmware)
GPU Family, in two letters acronym (-family)
SI - Southern Islands
CI - Sea Islands
KV - Kaveri
VI - Volcanic Islands
CZ - Carrizo
AI - Arctic Islands
RV - Raven
NV - Navi
Example result

$ kubectl describe node cluster-node-23
Name:               cluster-node-23
Roles:              <none>
Labels:             beta.amd.com/gpu.cu-count.64=1
                    beta.amd.com/gpu.device-id.6860=1
                    beta.amd.com/gpu.family.AI=1
                    beta.amd.com/gpu.simd-count.256=1
                    beta.amd.com/gpu.vram.16G=1
                    beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/hostname=cluster-node-23
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
                    node.alpha.kubernetes.io/ttl: 0
......

Limitations

While container scheduling via a label selector works for a heterogeneous cluster, it requires homogenous nodes. For example, you can have a node with just Fiji and just Vega10 in the same cluster but not a node that has both Fiji and Vg10 cards in it.

[the remark this answers is not in the document]
Comment (tizhou86) [reply docs/design/gpu-and-kunlun-support.md:70]: > Did you look at the node labeler work AMD did for it k8 device plug-in https://github.com/RadeonOpenCompute/k8s-device-plugin/blob/master/cmd/k8s-node-labeller/README.md
> 
> The tool automatically labels nodes with GPU properties if a node has one or more AMD GPU installed. Aka so you can support different GPU/Devices in the cluster ( but you need homogenous nodes) and make it easier to select the appropriate GPU in a node in the cluster
> 
> Deployment
> 
> The Labeller needs to be run on all the nodes that are equipped with AMD GPU. It creates a Kubernetes DaemonSet, which runs a copy of a pod on all (or some) Nodes in the cluster. An example configuration is available here.
> 
> The Labeller currently creates node label for the following AMD GPU properties:
> 
> Device ID (-device-id)
> VRAM Size (-vram)
> Number of SIMD (-simd-count)
> Number of Compute Unit (-cu-count)
> Firmware and Feature Versions (-firmware)
> GPU Family, in two letters acronym (-family)
> SI - Southern Islands
> CI - Sea Islands
> KV - Kaveri
> VI - Volcanic Islands
> CZ - Carrizo
> AI - Arctic Islands
> RV - Raven
> NV - Navi
> Example result
> 
> $ kubectl describe node cluster-node-23
> Name: cluster-node-23
> Roles:
> Labels: beta.amd.com/gpu.cu-count.64=1
> beta.amd.com/gpu.device-id.6860=1
> beta.amd.com/gpu.family.AI=1
> beta.amd.com/gpu.simd-count.256=1
> beta.amd.com/gpu.vram.16G=1
> beta.kubernetes.io/arch=amd64
> beta.kubernetes.io/os=linux
> kubernetes.io/hostname=cluster-node-23
> Annotations: kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
> node.alpha.kubernetes.io/ttl: 0
> ......
> 
> Limitations
> 
> While container scheduling via a label selector works for a heterogeneous cluster, it requires homogenous nodes. For example, you can have a node with just Fiji and just Vega10 in the same cluster but not a node that has both Fiji and Vg10 cards in it.

sounds good, i'll figure out the details of your implementation, thanks.

**Pull request closed without merging**